evap/staff/views.py

```diff
@@ -1330,17 +1330,19 @@ def notify_reward_points(grantings, **_kwargs):
         Evaluation, Contribution, formset=ContributionFormset, form=ContributionForm, extra=1 if editable else 0
     )
 
-    evaluation_form = EvaluationForm(request.POST or None, instance=evaluation, semester=evaluation.course.semester)
+    operation = request.POST.get("operation")
+
+    if request.method == "POST" and operation not in ("save", "approve"):
+        raise SuspiciousOperation("Invalid POST operation")
+
+    evaluation_form = EvaluationForm(
+        request.POST or None, instance=evaluation, semester=evaluation.course.semester, operation=operation
```

Comment:
random consideration: A narrower, more precise interface would be passing `require_main_language: bool` instead of passing an operation string that implicitly allows to infer that bool